app/src/main/java/com/firebase/uidemo/database/ChatActivity.java

```diff
@@ -131,26 +136,50 @@ public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
     }
 
     private void attachRecyclerViewAdapter() {
-        Query lastFifty = mChatRef.limitToLast(50);
-        mAdapter = new FirebaseRecyclerAdapter<Chat, ChatHolder>(
-                Chat.class, R.layout.message, ChatHolder.class, lastFifty) {
+        mAdapter = new FirebaseIndexRecyclerAdapter<Chat, ChatHolder>(
```

Comment:
@samtstern Hmmm, I didn't think about people copypasta-ing code straight from `ChatActivity`... If that's the case, then I agree with you that the code should be as simple as possible and I like the idea of a separate activity for the index stuff. What do you think of 758b69239048050f163956c3c1e56b1e12edd76d?